postgres/tests/test_statements.py

```diff
@@ -1594,3 +1610,43 @@ def test_pg_stat_statements_max_warning(
     check.check(dbm_instance)
 
     assert check.warnings == expected_warnings
+
+
+# This test relies on replica so we need PG>10
+@requires_over_10
+def test_pg_stat_statements_dealloc(aggregator, integration_check, dbm_instance_replica2):
```

Comment:
Since the dealloc code requires >v14, should we use the `@requires_over_14` decorator instead and skip the checks within the test?

Reply:
I would (eventually) like to make sure that we don't generate a dealloc metric for <v14 (to make sure we don't generate an empty metric in those case for example). 
The only way to do this seems to be through `aggregator.assert_all_metrics_covered` which will require some test refactoring before.